### ISSUE TITLE
Consider using `balanced_accuracy` as default scoring

### ISSUE THREAD

**Issue by elcorto** [psteinb/c2st#4]
The default `scoring='accuracy'` can result in unexpected results (return value not close to 0.5) if sample sizes are different because the classifier is trained on unbalanced data, e.g.

```py
>>> for ny in arange(100, 1100, 100):
...     print(f"{ny=} {compare(randn(ny,90), randn(500,90), scoring='accuracy')=}")
ny=100 compare(randn(ny,90), randn(500,90), scoring='accuracy')=array([0.83166665], dtype=float32)
ny=200 compare(randn(ny,90), randn(500,90), scoring='accuracy')=array([0.71], dtype=float32)
ny=300 compare(randn(ny,90), randn(500,90), scoring='accuracy')=array([0.61875], dtype=float32)
ny=400 compare(randn(ny,90), randn(500,90), scoring='accuracy')=array([0.5188889], dtype=float32)
ny=500 compare(randn(ny,90), randn(500,90), scoring='accuracy')=array([0.49], dtype=float32)
ny=600 compare(randn(ny,90), randn(500,90), scoring='accuracy')=array([0.4990909], dtype=float32)
ny=700 compare(randn(ny,90), randn(500,90), scoring='accuracy')=array([0.56416667], dtype=float32)
ny=800 compare(randn(ny,90), randn(500,90), scoring='accuracy')=array([0.5953846], dtype=float32)
ny=900 compare(randn(ny,90), randn(500,90), scoring='accuracy')=array([0.63285714], dtype=float32)
ny=1000 compare(randn(ny,90), randn(500,90), scoring='accuracy')=array([0.6626667], dtype=float32)

>>> for ny in arange(100, 1100, 100):
...     print(f"{ny=} {compare(randn(ny,90), randn(500,90), scoring='balanced_accuracy')=}")
ny=100 compare(randn(ny,90), randn(500,90), scoring='balanced_accuracy')=array([0.4990196], dtype=float32)
ny=200 compare(randn(ny,90), randn(500,90), scoring='balanced_accuracy')=array([0.50277776], dtype=float32)
ny=300 compare(randn(ny,90), randn(500,90), scoring='balanced_accuracy')=array([0.48938507], dtype=float32)
ny=400 compare(randn(ny,90), randn(500,90), scoring='balanced_accuracy')=array([0.51276666], dtype=float32)
ny=500 compare(randn(ny,90), randn(500,90), scoring='balanced_accuracy')=array([0.4938446], dtype=float32)
ny=600 compare(randn(ny,90), randn(500,90), scoring='balanced_accuracy')=array([0.4878621], dtype=float32)
ny=700 compare(randn(ny,90), randn(500,90), scoring='balanced_accuracy')=array([0.5029723], dtype=float32)
ny=800 compare(randn(ny,90), randn(500,90), scoring='balanced_accuracy')=array([0.49580008], dtype=float32)
ny=900 compare(randn(ny,90), randn(500,90), scoring='balanced_accuracy')=array([0.49264896], dtype=float32)
ny=1000 compare(randn(ny,90), randn(500,90), scoring='balanced_accuracy')=array([0.49767256], dtype=float32)
```

An extended version of this might also go into an example notebook, e.g. plotting the return value vs `ny` (expected: stay around 0.5) for various `scoring` settings from https://scikit-learn.org/stable/modules/model_evaluation.html#common-cases-predefined-values .

**Comment by elcorto** [comment]
Instead or in addition, one might change the default CV method to `sklearn.model_selection.StratifiedKFold`.

**Comment by psteinb** [comment]
Both changes are well motivated. Thanks for the suggestion. I am happy to implement it when the time capacity arises. If you'd like to send a PR, feel free to do so.

To be fair, I used `c2st` so far with equal sized samples and hence didn't consider to add an account for unbalanced training data. I'll keep the issue open until a fix lands.

**Comment by elcorto** [comment]
A quick test shows that `StratifiedKFold` doesn't seem to help much, so I wouldn't make that default w/o further testing. `scoring="balanced_accuracy"` is the best bet so far.

```py
>>> cv=StratifiedKFold(n_splits=5, random_state=1, shuffle=True)

>>> for nx in arange(100, 1100, 100)
...     print(f"{nx=} {c2st(randn(nx,90), randn(500,90), scoring='accuracy', cv=cv)}")
nx=100 [0.8333333]
nx=200 [0.7114286]
nx=300 [0.62875]
nx=400 [0.53555554]
nx=500 [0.521]
nx=600 [0.5190909]
nx=700 [0.55583334]
nx=800 [0.6076923]
nx=900 [0.6357143]
nx=1000 [0.66533333]

>>> for nx in arange(100, 1100, 100):
...     print(f"{nx=} {c2st(randn(nx,90), randn(500,90), scoring='balanced_accuracy', cv=cv)}")
nx=100 [0.5]
nx=200 [0.496]
nx=300 [0.517]
nx=400 [0.523]
nx=500 [0.496]
nx=600 [0.5056667]
nx=700 [0.50885713]
nx=800 [0.49875]
nx=900 [0.5047778]
nx=1000 [0.5005]
```

This was done with the code version mentioned in #8 where we can pass `cv`.

**Comment by elcorto** [comment]
I added a param study to turn the numbers above into plots, see [here](https://github.com/elcorto/c2st/blob/feature-param-study/rsc/param_study/hyper.ipynb), end of notebook. This last study concerns the `scoring` kwd and the CV method discussed in this issue.

**Comment by psteinb** [comment]
This is a wonderful study. Thanks for sharing. Feel free to send a PR with the balanced accuracy changes at any point. To me, the notebooks can also be included in this repo.

**Comment by elcorto** [comment]
Thanks. Balanced accuracy is covered in #10. 

The notebook can't go in at this point since it rellies on changes proposed in #8 #9 and implemented in [this branch](https://github.com/elcorto/c2st/tree/feature-refactor-api) so you might want to have a look at this first :) Unrelated to the notebook, this branch also covers #2 (without breaking the current API) and #3.
